### PR TITLE
ci: introduce proto breakage detector

### PR DESCRIPTION
**Pull request by cmwaters**
Still thinking about whether we should enable this on `main`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,6 +58,11 @@ proto-lint:
 	@$(DOCKER_BUF) lint --error-format=json
 .PHONY: proto-lint
 
+## proto-check-breaking checks if their is any breaking change to the protos
+proto-check-breaking:
+	@$(DOCKER_BUF) breaking --against $(HTTPS_GIT)#branch=v1.x
+.PHONY: proto-lint
```
Comment (rootulp) [inline Makefile:64]: [blocking]

```suggestion
.PHONY: proto-check-breaking
```

File: Makefile [hunk 2]
```diff
@@ -57,6 +58,11 @@ proto-lint:
 	@$(DOCKER_BUF) lint --error-format=json
 .PHONY: proto-lint
 
+## proto-check-breaking checks if their is any breaking change to the protos
```
Comment (rootulp) [inline Makefile:61]: Currently `make help` results in:

```
$ make help
 Choose a command run in celestia-app:
  help                                                                        Get more info on make commands.
  build                                                                       Build the celestia-appd binary into the ./build directory.
  install                                                                     Build and install the celestia-appd binary into the $GOPATH/bin directory.
  mod                                                                         Update go.mod.
  mod-verify                                                                  Verify dependencies have expected content.
  proto-gen                                                                   Generate protobuf files. Requires docker.
  proto-lint                                                                  Lint protobuf files. Requires docker.
  proto-check-breaking checks if their is any breaking change to the protos
```

but if we add a colon, this prints nicer

```suggestion
## proto-check-breaking: Check if there are any breaking change to protobuf definitions.
```

```
$ make help
 Choose a command run in celestia-app:
  help                   Get more info on make commands.
  build                  Build the celestia-appd binary into the ./build directory.
  install                Build and install the celestia-appd binary into the $GOPATH/bin directory.
  mod                    Update go.mod.
  mod-verify             Verify dependencies have expected content.
  proto-gen              Generate protobuf files. Requires docker.
  proto-lint             Lint protobuf files. Requires docker.
  proto-check-breaking   Check if there are any breaking change to protobuf definitions.
```

File: .github/workflows/protobuf.yml [hunk 3]
```diff
@@ -32,10 +32,9 @@ jobs:
...
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #     - name: check-breakage
-  #       run: make proto-check-breaking
+  breakage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check-breakage
```
Comment (rootulp) [inline .github/workflows/protobuf.yml:39]: [nit] can the step name match the Makefile task name so it's easy for unfamiliar maintainers to connect the GH action in the list of CI tasks to the Makefile task? Note, the suggestion is consistent with proto-lint and proto-gen above.

```suggestion
  proto-check-breaking:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: proto-check-breaking
```

File: .github/workflows/protobuf.yml [hunk 4]
```diff
@@ -32,10 +32,9 @@ jobs:
...
+        run: make proto-check-breaking
```
Comment (rootulp) [inline .github/workflows/protobuf.yml:40]: [blocking] I can't comment on this line directly but can we either update or delete the top line of this file. IMO seems safe to delete.

Ref: https://github.com/celestiaorg/celestia-app/pull/2671/files#diff-a44d106fd1ff0127f62ef1b829e39ff1e4697db877213026cd87e57e7728161fR1

File: Makefile [hunk 5]
```diff
@@ -57,6 +58,11 @@ proto-lint:
...
+proto-check-breaking:
+	@$(DOCKER_BUF) breaking --against $(HTTPS_GIT)#branch=v1.x
```
Comment (rootulp) [inline Makefile:63]: [question] the first time I ran this, it seems to have timed out. Have you encountered this?

```
$ make proto-check-breaking
Unable to find image 'bufbuild/buf:latest' locally
latest: Pulling from bufbuild/buf
579b34f0a95b: Pull complete
a8989812ba48: Pull complete
cb12522669ac: Pull complete
Digest: sha256:bd804cf64c2d220dd5503d008c49d33cb7a4c6ada9b952170a6f516e45c97531
Status: Downloaded newer image for bufbuild/buf:latest










Failure: the server hosted at that remote is unavailable.
make: *** [proto-check-breaking] Error 1
```

The second time, I see no output... proposal to include a line that informs the developer that the task is actually running.

```
$ make proto-check-breaking
```

```suggestion
proto-check-breaking:
	@echo "--> Checking if Protobuf definitions have any breaking changes"
	@$(DOCKER_BUF) breaking --against $(HTTPS_GIT)#branch=v1.x
```

File: Makefile (hunk 2 above)
Comment (cmwaters) [reply Makefile:61]: Ah I didn't know about this - thanks

**Comment by cmwaters** [comment]
> do we intend on making this a required CI check after it merges?

Yeah I think we should

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Comment by KJ1403** [comment]
Your Choice

**Pull request merged**